crates/runtime/src/embeddings/vector_search.rs

```diff
@@ -140,7 +182,7 @@ impl SearchRequest {
         text: String,
         datasets: Option<Vec<String>>,
         limit: usize,
-        where_cond: Option<String>,
+        where_cond: Option<Expr>,
```

Comment:
nit: add docstring that we expect this `where_cond` to be validated/sanitised. And `TryFrom` should be used otherwise.